[CPU][Emitters] Add more informative exception messages in emitters

a-sidorova opened the pull request:
### Details:
 - *Added new files `name_` to `jit_emitter` to make exception messages more informative: now each emitter throws exception with prefix `The emitter <name_> ...`.*

- *The example:*
 ```
./bin/intel64/Release/ov_cpu_func_tests --gtest_filter='*smoke_Snippets*'
[ RUN      ] smoke_Snippets_Eltwise/Add.CompareWithRefImpl/IS[0]=[]_TS[0]=((1.16.29.31))_IS[1]=[]_TS[1]=((1.16.29.1))_T=f32_#N=1_#S=1_targetDevice=CPU

MEM_USAGE=773124KB
src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp:90: Failure
Exception from src/inference/src/core.cpp:99:
[ GENERAL_ERROR ] Check 'false' failed at src/plugins/intel_cpu/src/emitters/plugin/x64/jit_emitter.cpp:182:
The emitter jit_nop_emitter throws test exception


unknown file: Failure
C++ exception with description "Exception from src/inference/src/compiled_model.cpp:35:
CompiledModel was not initialized.
" thrown in the test body.
[  FAILED  ] smoke_Snippets_Eltwise/Add.CompareWithRefImpl/IS[0]=[]_TS[0]=((1.16.29.31))_IS[1]=[]_TS[1]=((1.16.29.1))_T=f32_#N=1_#S=1_targetDevice=CPU, where GetParam() = (({}, { { 1, 16, 29, 31 } }), ({}, { { 1, 16, 29, 1 } }), f32, 1, 1, "CPU") (12 ms)
 ```
 
 #### Updated (17/01):
  - *Added macros `OV_CPU_JIT_EMITTER_THROW` and `OV_CPU_JIT_EMITTER_ASSERT` (that are declared in [emitter.hpp](https://github.com/a-sidorova/openvino/blob/81b19d76ed15a44876b08fc79d56a5a14b02cb0f/src/plugins/intel_cpu/src/emitters/plugin/x64/jit_emitter.hpp#L42)) - inserted name of jit emitter with namespaces (using `__PRETTY_FUNC__` with GNU and `__FUNCSIG__` without GNU) and text of exception.*
  - *The example of exception in Release build on Ubuntu:*
```
src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp:90: Failure
Exception from src/inference/src/core.cpp:99:
[ GENERAL_ERROR ] Exception from src/plugins/intel_cpu/src/emitters/plugin/x64/jit_eltwise_emitters.cpp:60:
ov::intel_cpu::jit_add_emitter: Test throw from jit_add_emitter
```
  - *The example of exception in Release build on Windows:*
```
src\tests\functional\shared_test_classes\src\base\ov_subgraph.cpp(90): error: Exception from src\inference\src\core.cpp:102:

src\inference\src\ie_common.cpp:71 [ GENERAL_ERROR ] Exception from src\plugins\intel_cpu\src\emitters\plugin\x64\jit_eltwise_emitters.cpp:60:
ov::intel_cpu::jit_add_emitter: Test assert from jit_add_emitter
```
  - *The example of assertion in Release build on Ubuntu:*
```
src/tests/functional/shared_test_classes/src/base/ov_subgraph.cpp:90: Failure
Exception from src/inference/src/core.cpp:99:
[ GENERAL_ERROR ] Check '(false)' failed at src/plugins/intel_cpu/src/emitters/plugin/x64/jit_eltwise_emitters.cpp:60:
ov::intel_cpu::jit_add_emitter: Test assert from jit_add_emitter
```
  - *The example of assertion in Release build on Windows:*
```
src\tests\functional\shared_test_classes\src\base\ov_subgraph.cpp(90): error: Exception from src\inference\src\core.cpp:102:

src\inference\src\ie_common.cpp:71 [ GENERAL_ERROR ] Check '(false)' failed at src\plugins\intel_cpu\src\emitters\plugin\x64\jit_eltwise_emitters.cpp:60:
ov::intel_cpu::jit_add_emitter: Test assert from jit_add_emitter
```

### Tickets:
 - *N/A*

dmitry-gorokhov commented:
As we agreed offline need to rework solution using pre-compiler defines instead of manually specifying names

a-sidorova commented:
@dmitry-gorokhov I added macros that uses pre-compiler defines as we discussed. Could you please take a look one more time? Thanks a lot!

I've attached the examples of test exceptions/assertions to the PR description for Ubuntu (GNU) and Windows.

a-sidorova commented:
@dmitry-gorokhov as discussed offline, I extracted the defines and the function `jit_emitter_pretty_name` to the separate files `utils.hpp` and added finding handler in the `jit_emitter_pretty_name` to avoid undefined behavior. Could you please take a look again? Thanks!
